### DISCUSSION TITLE
Auto-reconnect with virsh?

### DISCUSSION THREAD

**Discussion by sweharris** [bstansell/conserver#84] [Q&A]
I have a number of consoles pointing to virtual machines

eg

```
console my_vm {
        type exec;
        exec "/usr/bin/virsh -c qemu+ssh://vmserver/system console my_vm";
}
```

This works well; I can "console my_vm".   And because I have a logfile enabled it means that if a VM reboots for any reason then I can see the console logs.  Perfect!

Except... if the VM is terminated for any reason (eg server reboot for patching) then the virsh connection terminates.
```
error: Disconnected from qemu+ssh://vmserver/system due to end of file
```
And it doesn't reconnect until I manually `console my_vm` and hit return; that causes conserver to detect the client has died (generating a "Console down" message), and to respawn (generating a "Console up")

eg
```
[-- MARK -- Wed Sep 14 07:00:00 2022]
error: Disconnected from qemu+ssh://penfold/system due to end of file
[-- MARK -- Wed Sep 14 08:00:00 2022]
...
[-- MARK -- Tue Nov  8 12:00:00 2022]
[-- sweh@localhost attached -- Tue Nov  8 12:52:57 2022]

[-- Console down -- Tue Nov  8 12:52:59 2022]
[-- Console up -- Tue Nov  8 12:52:59 2022]
Connected to domain my_vm
```
There's an almost 2 month gap in my logs as a result!

Can conserver auto reconnect in this scenario?  Or am I better off writing a wrapper shell script around virsh that will retry?

**Comment by pcfe** [comment]
I came across this when looking for `virsh console …` examples.

While not perfect, reconnect tries are by default every minute, the addition of

```
options autoreinit;
```

might serve your needs.

**Comment by sweharris** [comment]
Hmm, thanks for that.  The manpage says  "Default is autoreinit." so I assume it's set by default?  But I'll definitely set it (just in case the manpage is wrong) and see if it helps :-)

Thanks!
